### PR TITLE
Setup frontend infrastructure

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

#50

Sets up frontend infrastructure for developing the CryoET Data Portal website. The tooling included in this PR are:

- Remix: React web framework for handling SSR, routing, and data fetching
- ESlint: Tool for checking code quality and verifying code meets our standards
- Prettier: Code formatter that checks and formats code using a unified coding style
- Jest: Test runner for running unit tests
- React Testing Framework: Library for testing react components
- Material UI: Component framework for using Material Design components in React
- Science Design System: Component framework by the SDS team
- Stylelint: Linter for SCSS code
- Tailwind: Tooling for styling HTML using utility classes
- pnpm: Fast package manager with monorepo capability 

## Data Portal

image

### CODE REVIEW

File: frontend/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# CryoET Data Portal Frontend Monorepo
+
+This directory contains the entire source code for the CryoET Data Portal Frontend, organized as a monorepo containing several packages that have different responsibilities:
```
Comment (kne42) [inline frontend/README.md:3]: love the detailed README 💯

File: frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "scripts": {
+    "preinstall": "npx only-allow pnpm",
+    "build": "pnpm --parallel build",
+    "clean": "rm -rf .cache && pnpm --parallel clean",
+    "clean:node-modules": "rm -rf node_modules && rm -rf packages/*/node_modules",
+    "dev": "pnpm --parallel dev",
+    "dev:data-portal": "pnpm --filter data-portal dev",
+    "dev:graphiql": "pnpm --filter graphiql dev",
+    "lint": "run-p lint:*",
+    "lint:packages": "pnpm --parallel lint",
```
Comment (kne42) [inline frontend/package.json:12]: what does pnpm lint for exactly? I want to add prettier to our [pre-commit config](https://pre-commit.com/) but it looks like I would have to do [hacky things for pnpm stuff](https://stackoverflow.com/questions/76268277/pnpm-for-pre-commit-custom-hook)

**Review by kne42 — APPROVED** [review]

File: frontend/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+services:
+  data-portal:
+    image: cryoet-data-portal/data-portal
+    build:
+      context: .
+      dockerfile: packages/data-portal/Dockerfile
+    ports:
+      - 8080:8080
+
+  graphiql:
```
Comment (manasaV3) [inline frontend/docker-compose.yml:12]: Could we leverage the Hasura admin console in place of the graphiql?

File: frontend/packages/data-portal/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+FROM node:20.8-alpine as build
+
+COPY . /app
```
Comment (manasaV3) [inline frontend/packages/data-portal/Dockerfile:3]: Just curious, do we need files related to linting to be a part of the build, or could we add those to the `.dockerignore`?

**Review by manasaV3 — COMMENTED** [review]
Really cool stuff.. 🔥

File: frontend/.gitignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.env
```
Comment (manasaV3) [inline frontend/.gitignore:1]: Curious about why these can't be a part of the parent `.gitignore` file.

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**